### PR TITLE
Feature/cxspa 4968 display from quote cart

### PR DESCRIPTION
**Pull request by ChristophHi**

### CODE REVIEW

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 1]
```diff
@@ -38,19 +51,19 @@ export class ConfigureCartEntryComponent {
...
-    }
+    return this.getOwnerTypeInternal(true);
+  }
+
+  /**
+   * Determines owner for an entry (can be part of cart, order or quote)
+   * @param cartCode Code of active cart
+   * @returns - an owner type
+   */
+
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts]:
```suggestion
```
just remove empty line between comment and code

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 2]
```diff
@@ -60,39 +73,77 @@ export class ConfigureCartEntryComponent {
    * @returns - an entry key
    */
   getEntityKey(): string {
+    return this.getEntityKeyInternal(true);
+  }
+  /**
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:78]:
```suggestion
 
   /**
```
empty line mising

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 3]
```diff
@@ -60,39 +73,77 @@ export class ConfigureCartEntryComponent {
...
         return this.cartEntry.quoteCode;
+      } else {
+        return this.cartEntry.savedCartCode;
       }
-      throw new Error(
-        ConfigureCartEntryComponent.ERROR_MESSAGE_ENTRY_INCONSISTENT
-      );
     } else {
       return undefined;
     }
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:140]: it seems like that this method is not used anymore -  do we keep it just for backward compatibility?

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 4]
```diff
@@ -60,39 +73,77 @@ export class ConfigureCartEntryComponent {
...
+   * Verifies whether the cart entry has an order code, retrieves a composed owner ID
+   * and concatenates a corresponding entry number.
+   * @param cartCode Code of active cart
+   * @returns - an entry key
+   */
+  getEntityKeyKnowingActiveCart(cartCode: string): string {
+    return this.getEntityKeyInternal(this.isActiveCart(cartCode));
+  }
+
+  protected isActiveCart(cartCode: string): boolean {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:88]: i find this method name confusing, aren't we checking whether we are in the context of **saved cart**, so maybe isSavedCartActive()?

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 5]
```diff
@@ -60,39 +73,77 @@ export class ConfigureCartEntryComponent {
...
+    return cartCode === this.cartEntry.savedCartCode;
+  }
+
+  protected getOwnerTypeInternal(
+    readOnlyCartIsActive: boolean
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:93]: so whats a readOnlyCart in this context?
i see its filled with result isActiveCart()

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts [hunk 6]
```diff
@@ -242,53 +242,42 @@ export class VariantConfiguratorOccAdapter
...
+        'readVariantConfigurationOverviewForOrderEntry',
+        {
+          urlParams: {
+            userId: parameters.userId,
+            orderId: parameters.orderId,
+            orderEntryNumber: parameters.orderEntryNumber,
+          },
+        }
+      );
+    }
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts:280]: all calls to buildUrl are quite similar. They only differ in the enpoint. So i suggest to only determine the enpoint within the if-else statement and then call  buildUrl afterwards.

File: feature-libs/quote/components/items/quote-items.component.html [hunk 7]
```diff
@@ -16,23 +16,37 @@
...
+          quoteDetails.cartId !== undefined &&
+          (quoteCartReadOnly$ | async) as quoteCartReadOnly
+        "
+        [cxOutlet]="cartOutlets.CART_ITEM_LIST"
+        [cxOutletContext]="{
+          items: quoteCartReadOnly.entries ?? [],
+          readonly: true
+        }"
+      >
+      </ng-template>
```
Comment (Uli-Tiger) [inline feature-libs/quote/components/items/quote-items.component.html]: i would move this just before the previous template, as it relates more to first case, then the second case
so we have

1. quote case
2. saved cart case (new)
3. active cart case

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]
see my comments

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 5 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:93]: I mean from the quote perspective it ts the readOnlyQuoteCart case - but from a config perspective we are in the savedCart scenario, which will also solve our 'readOnlyQuoteCart ' case. Still i would name this more with saved cart in mind-

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts (hunk 6 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts:280]:
```suggestion
    let endpoint;
    if (ownerType === CommonConfigurator.OwnerType.QUOTE_ENTRY) {
      endpoint = 'readVariantConfigurationOverviewForQuoteEntry';
    } else if (ownerType === CommonConfigurator.OwnerType.SAVED_CART_ENTRY) {
      endpoint = 'readVariantConfigurationOverviewForSavedCartEntry';
    } else {
      endpoint = 'readVariantConfigurationOverviewForOrderEntry';
    }
    let url = this.occEndpointsService.buildUrl(endpoint, {
      urlParams: {
        userId: parameters.userId,
        orderId: parameters.orderId,
        orderEntryNumber: parameters.orderEntryNumber,
      },
    });
```

**Comment by Uli-Tiger** [comment]
when i follow the steps from the task, so basically:
- Configure the screwdriver, set to professional
- Request quote for this cart
- Change config in quote and set to standard
- Click 'new cart'
- Call up previous quote
- Entry summary shows standard correctly,
- but if i try ti display the entries config i get a cart not found error!

using the mock as engine.
- 

i get a cart not found exception
image

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 3 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:140]: Yes, that was my idea. But I am currently preparing a new idea how to provide the document context to the outlets that will make the activeCartFacade obsolete, meaning most of the changes to this component most likely are not needed => we will use this method again in a simplified form

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 4 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:88]: No. This is really about checking if we are in active cart mode

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 5 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:93]: 'readOnlyCartIsActive' means: the readOnly cart we are dealing with here is the active cart.
But please note as I said above: Probably we will get rid of activeCartFacade anyhow and don't need that coding.

File: feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts (hunk 6 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/occ/variant/variant-configurator-occ.adapter.ts:280]: No. The url parameters are different

**Comment by ChristophHi** [comment]
@Uli-Tiger I cannot reproduce the issue that you mentioned yesterday, tried with CPS and then with mock on https://team-server.tiger-ci.only.sap:17002. Could you try again and maybe give me more hints on how to reproduce? Does it only happen on a local commerce instance?

**Review by Uli-Tiger — APPROVED** [review]
I rested to day and was not able to reproduce the cart not found exception. not sure how and why this happened yesterday,

**Pull request merged**